src/lib/cached/cached.ml

```diff
@@ -170,24 +191,61 @@ let run
               None )
   with
   | Some x ->
-      return x
+      return (x, `Cache_hit)
   | None -> (
       Core.printf
         "Could not load %s from the following path:\n\
         \ \n\
          %s\n\
          %s\n\
+         %s\n\
         \ \n\
-        \ Trying the autogen path %s...\n"
-        name manual_install_path brew_install_path autogen_path ;
-      let base_path = base_path autogen_path in
-      match%bind With_components.load load ~base_path with
+        \ Trying to hit the s3 bucket...\n"
+        name manual_install_path brew_install_path s3_install_path ;
+      (* Attempt load from s3 *)
+      let open Cohttp_async in
+      let open Deferred.Let_syntax in
+      match%bind
+        let open Deferred.Or_error.Let_syntax in
+        let%bind () =
+          Deferred.Result.join
+          @@ Monitor.try_with (fun () ->
+                 let open Deferred.Let_syntax in
+                 let%bind resp, body =
+                   Client.get
+                     (Uri.of_string (base_path "todo/path/to/s3/bucket"))
+                 in
+                 let body_pipe = Body.to_pipe body in
```

Comment:
Why do we need to pipe this? What does the response look like? Can't we just store the response we got into a file

Reply:
I use the pipe to store it to a file, is there a more direct way that you're thinking of?

Reply:
`Cohttp_async.Body.to_string`

Reply:
My understanding is that `to_string` will allocate a string for the entire response contents which is okay for verification keys, but proving keys are 500MB (I think this will crash the OCaml process). The pipe will ensure that bits of the data are written to the file as they come in over the network and we never have to allocate the whole thing in a big string.